ansible/roles/k8s_master/tasks/init_master_leader.yml

```diff
@@ -48,3 +48,18 @@
     dest: /home/ubuntu/.kube/config
     remote_src: yes
     owner: ubuntu
+ 
+- name: Helper for HAProxy to have copy of admin.conf
+  become: yes
+  fetch:
+    src: /etc/kubernetes/admin.conf
+    dest: /tmp/helper/admin.conf
+    flat: yes
+
+- name: Copy admin.conf file into HAPRoxy node to enable worker node join from VIP
+  delegate_to: "{{ haproxy_node }}"
+  become: yes
+  copy:
+    src: /tmp/helper/admin.conf
+    dest: /var/www/html/admin.conf
+    mode: '777'
```

Comment:
>777 [](start = 11, length = 3)

Everyone can write or del it, is it safe?

Reply:
I changed to 755 permissions. This was the most restrictive I could make it while ensuring that the SONiC DUT could still download the file properly. This is also in-line with the response here https://askubuntu.com/questions/451922/apache-access-denied-because-search-permissions-are-missing